shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -378,6 +444,36 @@ - (void)setupNotificationCenterObservers {
                object:nil];
 }
 
+- (void)setupKeyboardAnimationCurve {
+  _keyboardAnimationStops.reset([[NSMutableArray alloc] init]);
+
+  // Keyboard animation spring curve for iOS 16 and below
+  _keyboardSpringCurve.reset([[KeyboardSpringCurve alloc] initWithStiffness:1000
+                                                               dampingRatio:1
+                                                                       mass:3
+                                                            initialVelocity:0
+                                                           settlingDuration:0.5]);
```

Comment:
Pending🤔:

Still not very sure it is good for us to use this hardcoded parameters and cache it we app launches. Maybe better for us to dynamically get the animation curve and calculate instantly. That will be more flexible^_^.

Reply:
If we decide to hardcode the params for it, better use same param in `CASpringAnimation`, so that we don't need to compute manually (`dampingRatio` is not in it, we need to compute it ) ^_^
Here is iOS Demo code to show dynamically to get the animation params.
```swift
class ViewController: UIViewController {
    let tf = UITextField(frame: .zero)
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.tf.resignFirstResponder()
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = .white
        tf.backgroundColor = .red
        self.view.addSubview(tf)
        tf.frame = CGRect(x: 0, y: self.view.bounds.height - 100, width: self.view.bounds.width, height: 100)
        
        NotificationCenter.default.addObserver(forName: UIResponder.keyboardWillChangeFrameNotification, object: nil, queue: .main) { noti in
            self.tf.frame = CGRect(x: 0, y: self.view.bounds.height - 100, width: self.view.bounds.width, height: self.tf.frame.height + 1)
            let animation:CAAnimation? = self.tf.layer.animation(forKey: "position")
            if let spring:CASpringAnimation = animation as? CASpringAnimation {
                print(spring.damping) // 500
                print(spring.initialVelocity) // 0.0
                print(spring.mass) // 3.0
                print(spring.stiffness) // 1000
            }
        }
    }
}

```

Reply:
I didn't know this was possible! That's really cool. Will try to implement this tomorrow.

Reply:
Yes. That 's the way to calculate instantly ^_^. We can calculate every time, and if there are any changes for this spring, we will not change any code for it.

Reply:
But need to attection that, sometime, the animation will be nil or will not be a spring animation (When change keyboard type, such as language or change to a custom Keyboard ), I think we should use presentationLayer as a fallback to display, but most time, when open and close keyboard ,we will get the spring and using it to interpolate the keyboard animation (*^▽^*)(*^▽^*)

Reply:
Interesting - so you're suggesting we don't cache the spring animation stops on init and rather calculate every time?

I don't think calculating would be too expensive, so it could work without slowing anything down.

Reply:
Yes, I m suggesting don't cache stops. And calculate instantly. ^_^

Reply:
Okey dokey - will do tomorrow :)

Reply:
implemented the dynamic aspects, but haven't updated the spring curve formula to use damping vs dampingRatio